S3/S3.py

```diff
@@ -728,15 +728,17 @@ def content_type(self, filename=None, is_dir=False):
         return content_type
 
     def add_encoding(self, filename, content_type):
+        if not filename:
+            return False
         if 'charset=' in content_type:
            return False
         exts = self.config.add_encoding_exts.split(',')
         if exts[0]=='':
             return False
-        parts = filename.rsplit('.',2)
+        parts = filename.rsplit('.', 1)
         if len(parts) < 2:
             return False
-        ext = parts[1]
```

Comment:
now we have exactly two parts here and parts[-1] equals to parts[1]. I think parts[1] is more simple.